### ISSUE TITLE
[pgffor] \foreach breaks with arithmetics

### ISSUE THREAD

**Issue by PhelypeOleinik** [pgf-tikz/pgf#933]
**Brief outline of the bug**

Arithmetics in the start and end values of a `\foreach` used to work until `3.1.1` (TL 18), but somewhere between that and `3.1.5b` (TL 19) it stopped working, and it now breaks with `Illegal unit of measure (pt inserted)`.

**Minimal working example (MWE)**

```latex
\input pgffor
\def\nfreq{5}
\foreach \Y in {1,...,\nfreq-1} {\immediate\write16{\Y}}
\bye
```

Wrapping the whole thing around `\numexpr...\relax` is a workaround:

```latex
\input pgffor
\def\nfreq{5}
\foreach \Y in {1,...,\numexpr\nfreq-1\relax} {\immediate\write16{\Y}}
\bye
```

**Comment by marmotghost** [comment]
I do not think that this worked throughout all the versions between 3..1.1 and 3.1.5b. You can use the `parse` key. 
```
\foreach[parse=true] \Y in {1,...,\nfreq-1} {\Y}
```
A minimal LaTeX example.
```
\documentclass{article}
\usepackage{pgffor}
\begin{document}
\def\nfreq{5}
\foreach[parse=true] \Y in {1,...,\nfreq-1} {\Y}
\end{document}
```
However, the `\numexpr` stuff is in general superior for integers.

**Comment by hmenke** [comment]
Duplicate of #799. Won't fix because of #506.
